plugins/pur/core/src/main/java/com/pentaho/repository/importexport/StreamToJobNodeConverter.java

```diff
@@ -136,17 +145,36 @@ public IRepositoryFileData convert( final InputStream inputStream, final String
       Document doc = PDIImportUtil.loadXMLFrom( inputStream );
       if ( doc != null ) {
         jobMeta.loadXML( doc.getDocumentElement(), repository, null );
+        if ( jobMeta.hasMissingPlugins() ) {
+          KettleMissingPluginsException
+            missingPluginsException =
+            new KettleMissingPluginsException( getErrorMessage( jobMeta.getMissingEntries() ) );
+          throw new ConverterException( missingPluginsException );
+        }
         JobDelegate delegate = new JobDelegate( repository, this.unifiedRepository );
         delegate.saveSharedObjects( jobMeta, null );
         return new NodeRepositoryFileData( delegate.elementToDataNode( jobMeta ), size );
       } else {
         return null;
       }
-    } catch ( Exception e ) {
+    } catch ( IOException | KettleException e ) {
       return null;
```

Comment:
I can't understand why you are creating and throwing the ConverterException, to catch it here and do nothing with it. A null will be returned. At least, should be logged.

Reply:
I am not catching ConvertException anywhere! :P